src/shogun/evaluation/CrossValidation.cpp

```diff
@@ -26,22 +26,14 @@ CrossValidation::CrossValidation() : Seedable<MachineEvaluation>()
 }
 
 CrossValidation::CrossValidation(
-    std::shared_ptr<Machine> machine, std::shared_ptr<Features> features, std::shared_ptr<Labels> labels,
-    std::shared_ptr<SplittingStrategy> splitting_strategy, std::shared_ptr<Evaluation> evaluation_criterion)
-    : Seedable<MachineEvaluation>(
-          std::move(machine), std::move(features), std::move(labels), std::move(splitting_strategy), std::move(evaluation_criterion))
+	std::shared_ptr<Machine> machine, 
+	std::shared_ptr<SplittingStrategy> splitting_strategy, std::shared_ptr<Evaluation> evaluation_criterion)
+	: Seedable<MachineEvaluation>(
+          std::move(machine), std::move(splitting_strategy), std::move(evaluation_criterion))
```

Comment:
sure about the move here? The caller might still need the old references?

Reply:
Here it is fine, because he is passing the shared_ptr by value